elastic/kibana
# [Stateful sidenav] Set solution default route

In this PR I've set the `defaultRoute` for each of the 3 solutions based on the Space "solution" setting.
For that I had to modify the ui setting definition and add a new async `getValue()` handler that accepts a context object and returns the value of the setting.

In our case we needed the `KibanaRequest` to be passed in the context so we can dynamically return the default value based on the current active space (which is read from the request).

## How to test

* Add the following in `kibana.dev.yml`

```yml
xpack.cloud.id: "ftr_fake_cloud_id:aGVsbG8uY29tOjQ0MyRFUzEyM2FiYyRrYm4xMjNhYmM="
xpack.cloud.base_url: "https://cloud.elastic.co/"
xpack.cloud.deployment_url: "/deployments/deploymentId"
xpack.cloud_integrations.experiments.enabled: true
xpack.cloud_integrations.experiments.flag_overrides:
  "solutionNavEnabled": true
```

* Navigate to the Space management UI and create a new space
* Set the solution view to one of the solution
* Switch to that space using the space selector
* You should navigate to the correct default route based on the solution view setting

## Screenshots

<img width="527" alt="Screenshot 2024-07-09 at 11 18 52" src="https://github.com/elastic/kibana/assets/2854616/6b4beee6-41c7-4573-aefc-3c5ab6ac69ea">


<img width="1492" alt="Screenshot 2024-07-09 at 11 17 57" src="https://github.com/elastic/kibana/assets/2854616/c40f022d-fafa-4762-b798-3682a755b1f3">
<img width="1500" alt="Screenshot 2024-07-09 at 11 18 20" src="https://github.com/elastic/kibana/assets/2854616/d175df62-0db4-42e5-830f-15391cd4069b">

<img width="1498" alt="Screenshot 2024-07-09 at 11 17 33" src="https://github.com/elastic/kibana/assets/2854616/60c5c5df-d8a8-4141-b251-5685256a7c1a">
